src/include/migraphx/op/scatternd_op.hpp

```diff
@@ -28,44 +28,104 @@
 #include <migraphx/check_shapes.hpp>
 #include <migraphx/argument.hpp>
 #include <migraphx/par_for.hpp>
+#include <migraphx/ranges.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace op {
-
+/**
+ * @brief
+ * N-dimensional Scatter operations. This struct is parent class to ops which differ in what formula
+ * is used to reduce (combine old and new values of) the scattered value.  It is an implementation
+ * of the Onnx ScatterND operation (see
+ * https://github.com/onnx/onnx/blob/main/docs/Operators.md#ScatterND)
+ *
+ * @tparam Derived a type erasure parameter represents one of the child operations.
+ */
 template <class Derived>
 struct scatternd_op : op_name<Derived>
 {
+    /** Validate input shapes and return the correct output shape.  For Scatter ops, the output
+     * is the same shape as the data tensor (first input), but cast to a standard shape.
+     *
+     */
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3);
-        auto r         = inputs.front().lens().size();
-        auto q         = inputs.at(1).lens().size();
-        auto k         = inputs.at(1).lens().back();
-        auto ind_lens  = inputs.at(1).lens();
-        auto upd_lens  = inputs.back().lens();
-        auto data_lens = inputs.front().lens();
+        check_shapes{inputs, *this, true}.has(3);
+        auto sshape = inputs.front();
+
+        auto r = sshape.ndim();
+        auto q = inputs.at(1).ndim();
+        size_t k;
+        auto i_shape = inputs.at(1);
```

Comment:
Suggestion: If you have time you can change this to make it better. 
You can define 
`data_shape = inputs.front()`. 
`indices_shape = inputs.at(1);`
`updates_shape = inputs.back();` 

Once above at the beginning and use it everywhere instead of referencing `inputs.at(1)` or likewise.

Reply:
good idea.  Done